### ISSUE TITLE
model file

### ISSUE THREAD

**Issue by sxjzwq** [Cyanogenoid/vqa-counting#6]
Hi,

Can you share your trained model file?

**Comment by Cyanogenoid** [comment]
See here https://github.com/Cyanogenoid/vqa-counting/releases/tag/resources

**Comment by sxjzwq** [comment]
Thanks！Can you please also share your vocabulary (for both question and answer). It seems that there are some differences of your provided pre-trained model and the code.

I received following error when load your model:

While copying the parameter named "text.embedding.weight", whose dimensions in the model are torch.Size([13759, 300]) and whose dimensions in the checkpoint are torch.Size([16139, 300]).

**Comment by Cyanogenoid** [comment]
You are loading the trainval weights, right? Easiest way to get it working is to run `git checkout trainval`, then `python preprocess-vocab.py`. The issue is that the master branch creates the vocab from train, while the trainval model assumes the vocab to come from trainval.

Currently it always loads the vocab from the file system, but the correct vocab is stored in the .pth file. I'll change it so that it loads from the .pth when resuming.

**Comment by sxjzwq** [comment]
I see. Now it's fixed. But we find the test-dev results on VQA-2.0 of this model is  {"other": 57.21, "overall": 66.58, "number": 48.79, "yes/no": 82.27}}. Which is different from your presented accuracy. Any thing we missed?

Thanks!

**Comment by Cyanogenoid** [comment]
That's strange. Just to make sure, you are using the adaptive proposals (max 100 proposals per image) and not the 36 fixed proposals, correct? I just confirmed that I can exactly reproduce my results on the releases page using the pre-trained weights on trainval, so if that's not it, we'll have to look deeper.

**Comment by Cyanogenoid** [comment]
I'm getting almost exactly your results (within 0.04) when I use the 36 fixed proposals instead of 100 adaptive proposals during evaluation, so I'm pretty confident that that's the issue. Once you switch to the 100 adaptive proposals, you should be able to reproduce my numbers. Not sure where that tiny ~0.03 average difference comes from.


test-dev |  36 proposals
-- | --
other | 57.18
overall | 66.55
number | 48.81
yes/no | 82.23

This model is trained with adaptive 100 proposals and evaluated with 36 fixed proposals.

**Comment by ZhuFengdaaa** [comment]
I use 100 adaptive proposals but I got
```
[{"test-dev": {"other": 58.8, "overall": 67.9, "number": 51.03, "yes/no": 83.05}}]
```

**Comment by Cyanogenoid** [comment]
Yes, that is the expected result for the provided pre-trained model. sxjzwg told me that the issue is indeed because they were using 36 proposals instead of 100, so I'm closing this.
